### ISSUE TITLE
uAdmin Tutorial Part 7 - Introduction to API

### ISSUE THREAD

**Issue by MarErm27** [uadmin/uadmin#21]
Link: https://uadmin-docs.readthedocs.io/en/latest/tutorial/part7.html

Instruction: Establish a connection in the main.go to the API by using http.HandleFunc. It should be placed after the uadmin.Register and before the StartServer.
Changes for main.go file:
https://play.golang.org/p/jYJn2TKwk79

**Error:**
https://play.golang.org/p/W9uPsVvSMmC

Reason of the problem:
File: D:/MyProjects/src/github.com/uadmin/uadmin/register.go
Line 287: http.HandleFunc(RootURL+"api/", Handler(apiHandler))
**RootURL = "/"**
Here we have multiple registrations for /api/

**Workaround:**
In main.go replace
This:
http.HandleFunc("/api/", api.APIHandler)
To this:
http.HandleFunc("/admin/api/", api.APIHandler)

**Problem:**
This returns false:
https://play.golang.org/p/g2sBIdwjgAL

**Workaround:**
To make it work I've changed strings.HasSuffix() to strings.Contains() and it works.
At slack.com Money Sood and behappy recommended me to use strings.HasSuffix() and it also works.

To check if it works go to http://localhost:8080/admin/api/todo_list

My main.go file:
https://play.golang.org/p/wJKSDDbnKHM
My api.go file:
https://play.golang.org/p/_C2DjAJEVEU
**What http://localhost:8080/admin/api/todo_list returns:**
https://play.golang.org/p/AdaEwT3g4_l

**Comment by MarErm27** [comment]
Breakpoint in main.go:
![image](https://user-images.githubusercontent.com/56219126/72203718-45385780-3480-11ea-8098-858cd7fdcde2.png)
Breakpoint in register.go
![image](https://user-images.githubusercontent.com/56219126/72203725-68fb9d80-3480-11ea-8a2a-1c731bb383c2.png)

As you can see from breakpoints above uadmin.RootURL in main.go is not a same thing in register.go. Should they?

**Comment by Spotlight64** [comment]
Good day! Assigning uadmin.RootURL works on below version 0.2.0. For version 0.2.0 and above, you need to configure it in Settings to make it work.

![rooturlsetting](https://user-images.githubusercontent.com/49872771/72204894-bcd5a900-34b7-11ea-9aab-555cb77d772f.png)

On the other hand, we'll update the tutorial and references that follows the current standard. Thanks for reporting.

**Comment by MarErm27** [comment]
Thank you for the answer!
It's seems to me my settings looks different from yours. What may cause this?
![image](https://user-images.githubusercontent.com/56219126/72205427-ed0b5080-3493-11ea-9d72-405e10a22091.png)

I've change the value field and it makes everything in 7th article works. But when I tried to open http://localhost:8080/ and I got 404 error page not found. Is redirection from http://localhost:8080/ to http://localhost:8080/admin/ needed? It is not redirects automatically.

After I changed the Root URL setting, the search bar stopped to work. But filter does found it. Then I've tried to play with this setting once again and all pages become 404. After reloading app everything becomes perfect again.

Thanks for the feedback.
